request.go

```diff
@@ -202,18 +201,34 @@ func (req *Request) pack() (packet []byte, err error) {
 		byte(rid >> 8), byte(rid),
 	}
 
-	body, err = msgpack.Marshal(req.body)
+	enc := msgpack.NewEncoder(&h)
+	err = enc.EncodeMapLen(len(req.body))
 	if err != nil {
 		return
 	}
+	for k,v := range(req.body) {
+		err = enc.EncodeInt64(int64(k))
+		if err != nil {
+			return
+		}
+		switch vv:=v.(type) {
+		case uint32:
```

Comment:
I'm calling standard msgpack encoder for it. I'm just skipping some ineffeciency of `Encode` method.

By the way, this case may be hardwired here for performance:
- this integers are SpaceId, IndexId, IteratorType, Offset and Limit. There is no need to write them using smallest representatoon. They may be written as 4byte integers always.